pkg/controller/node/nodecontroller.go

```diff
@@ -525,6 +578,19 @@ func (nc *NodeController) monitorNodeStatus() error {
 				if err = markAllPodsNotReady(nc.kubeClient, node); err != nil {
 					utilruntime.HandleError(fmt.Errorf("Unable to mark all pods NotReady on node %v: %v", node.Name, err))
 				}
+
+				nodeUnreachableTaint := api.Taint{
+					Key:       unversioned.TaintNodeUnreachable,
+					Effect:    api.TaintEffectNoExecute,
+					AddedTime: nc.now(),
+				}
+				added, err := tryAddTaintToNode(nc.kubeClient, node.Name, nodeUnreachableTaint)
```

Comment:
Double checked here, though the node status change is recorded NodeNotReady, it is actually node unreachable (NodeReady status ConditionUnknown) here, so we should use TaintNodeUnreachable